src/hooks/useWalletConnection.ts

```diff
@@ -83,11 +83,28 @@ export const useWalletConnection = () => {
     key: LocalStorageKey.DydxAddress,
     defaultValue: undefined,
   });
-  const { data: dydxAccountGraz, isConnected: isConnectedGraz } = useAccountGraz();
-  const { signer: signerGraz } = useOfflineSignersGraz();
+  const { data: dydxAccountGraz, isConnected: isConnectedGraz } = useAccountGraz({
+    chainId: selectedDydxChainId,
+  });
+
   const { disconnectAsync: disconnectGraz } = useDisconnectGraz();
 
-  const dydxAddressGraz = dydxAccountGraz?.bech32Address;
+  const dydxAddressGraz = isConnectedGraz ? dydxAccountGraz?.bech32Address : undefined;
+
+  const getCosmosOfflineSigner = useCallback(
```

Comment:
I don't know if this easier but I think that graz has a [useOfflineSigners](https://graz.sh/docs/hooks/useOfflineSigners/) hook

Reply:
There was an issue using the useOfflineSigner hook, where there was constant prompts when disconnected, so this ended up changing
constext: https://dydx-team.slack.com/archives/C071QCEHM35/p1723735850192219